acir/src/circuit/opcodes.rs

```diff
@@ -166,15 +166,20 @@ impl std::fmt::Display for Opcode {
                     bits.last().unwrap().witness_index(),
                 )
             }
-            Opcode::Directive(Directive::Log(info)) => match info {
-                LogInfo::FinalizedOutput(output_string) => write!(f, "Log: {output_string}"),
-                LogInfo::WitnessOutput(witnesses) => write!(
-                    f,
-                    "Log: _{}..._{}",
-                    witnesses.first().unwrap().witness_index(),
-                    witnesses.last().unwrap().witness_index()
-                ),
-            },
+            Opcode::Directive(Directive::Log { is_trace, output_info }) => {
+                let is_trace_display = if *is_trace { "trace" } else { "println" };
+                match output_info {
+                    LogOutputInfo::FinalizedOutput(output_string) => {
+                        write!(f, "Log: {output_string}, log type {is_trace_display}")
+                    }
+                    LogOutputInfo::WitnessOutput(witnesses) => write!(
+                        f,
+                        "Log: _{}..._{}, log type: {is_trace_display}",
+                        witnesses.first().unwrap().witness_index(),
+                        witnesses.last().unwrap().witness_index(),
+                    ),
```

Comment:
We don't have a guarantee that these witnesses are in order/contiguous, do we?

e.g. for the program
```
fn main(x: pub Field, y: Field) -> [Field; 2];
    let z = 2*y
    let return_value = [z, x];
    std::println(return_value);
    return_value 
```

Reply:
ah you are right we technically don't. I don't know how I feel about printing the entire witness vec though. Perhaps it is ok as users will most likely only be using println's while debugging?

Reply:
Yeah, I'm not suggesting that we print out a full list as that would get messy for large arrays/strings fast. I'd be fine with just documenting in the code that this output is a small lie.